patches/src/main/java/org/jclouds/docker/compute/options/DockerTemplateOptions.java

```diff
@@ -0,0 +1,650 @@
+      this.cpuShares = cpuShares;
+      return this;
+   }
+
+   public DockerTemplateOptions env(Iterable<String> env) {
+      this.env = NullSafeCopies.copyWithNullOf(env);
+      return this;
+   }
+
+   public DockerTemplateOptions env(String...env) {
+      this.env = NullSafeCopies.copyWithNullOf(env);
+      return this;
+   }
+
+   /**
+    * Set port bindings between the Docker host and a container.
+    * <p>
+    * The {@link Map} keys are host ports number, and the value for an entry is the
+    * container port number. This is the same order as the arguments for the
+    * {@code --publish} command-line option to {@code docker run} which is
+    * {@code hostPort:containerPort}.
+    *
+    * @param portBindings the map of host to container port bindings
+    */
+   public DockerTemplateOptions portBindings(Map<Integer, Integer> portBindings) {
+      this.portBindings = NullSafeCopies.copyOf(portBindings);
+      return this;
+   }
+
+   /**
+    * Sets the networking mode for the container. Supported values are: bridge, host, and container:[name|id]
+    * @param networkMode
+    * @return this instance
+    */
+   public DockerTemplateOptions networkMode(@Nullable String networkMode) {
+      this.networkMode = networkMode;
+      return this;
+   }
+
+   /**
+    * Set extra hosts file entries for a container.
+    * <p>
+    * The {@link Map} keys are host names, and the value is an IP address that
+    * can be accessed by the container. This is the same order as the arguments for the
+    * {@code --add-host} command-line option to {@code docker run}.
+    *
+    * @param extraHosts the map of host names to IP addresses
+    */
+   public DockerTemplateOptions extraHosts(Map<String, String> extraHosts) {
+      this.extraHosts = NullSafeCopies.copyWithNullOf(extraHosts);
+      return this;
+   }
+
+   /**
+    * Set list of containers to mount volumes from onto this container.
+    *
+    * @param volumesFrom the list of container names
+    */
+   public DockerTemplateOptions volumesFrom(Iterable<String> volumesFrom) {
+      this.volumesFrom = NullSafeCopies.copyWithNullOf(volumesFrom);
+      return this;
+   }
+
+   public Map<String, String> getVolumes() { return volumes; }
+
+   public List<String> getDns() { return dns; }
+
+   public List<String> getVolumesFrom() { return volumesFrom; }
+
+   public String getHostname() { return hostname; }
+
+   public Integer getMemory() { return memory; }
+
+   public List<String> getEntrypoint() { return entrypoint; }
+
+   public List<String> getCommands() { return commands; }
+
+   public Integer getCpuShares() { return cpuShares; }
+
+   public List<String> getEnv() { return env; }
+
+   public Map<Integer, Integer> getPortBindings() { return portBindings; }
+
+   public String getNetworkMode() { return networkMode; }
+
+   public Map<String, String> getExtraHosts() { return extraHosts; }
+
+   public static class Builder {
```

Comment:
Not sure this is necessary as `DockerTemplateOptions` already has fluent methods, and the `Builder` methods below simple return a DTO, not a builder